### PR TITLE
Feature/GH#96 Added check to detect presence of wp_cache_flush()

### PR DESCRIPTION
**Pull request by Sidsector9**
Related #96

### CODE REVIEW

**Comment by Sidsector9** [comment]
@wp-cli/committers , I'm unable to understand the error due to which the builds are failing, can you point me where exactly did I go wrong?

**Comment by schlessera** [comment]
@Sidsector9 Behat is not recognizing the table you've provided as a table.

I suspect it is because of [broken indentation](https://github.com/wp-cli/doctor-command/pull/116/files#diff-b544c47a94e2a348c0d45d9c423efffcR16), check your table to make sure it uses the correct syntax.

Also, please remove the changes to the README file from this PR, such changes should go into a different PR instead.

**Comment by schlessera** [comment]
Ah, you can probably get rid of the README changes just by merging master branch again.

File: inc/checks/class-cache-flush.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+use WP_CLI;
+
+/**
+ * Detects the number of occurrences of the `wp_cache_flush()` function.
+ */
+class Cache_Flush extends Check {
+
+	public function run() {
+		$result = exec( 'grep -r "wp_cache_flush()" ' . WP_CONTENT_DIR . ' | wc -l' );
```
Comment (danielbachhuber) [inline inc/checks/class-cache-flush.php]: We should use the existing `runcommand\Doctor\Checks\File` filesystem iterator check in here.

Doing so will prevent iterating the filesystem for each check, instead only iterating the filesystem once.

Comment (Sidsector9) [reply inc/checks/class-cache-flush.php]: @danielbachhuber , I've updated this with iterator, let me know if this is one correctly.

File: inc/checks/class-cache-flush.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		// Path to wp-content directory.
+		$wp_content_dir = defined( 'WP_CONTENT_DIR' ) ? WP_CONTENT_DIR : ABSPATH . 'wp-content';
+		$directory = new RecursiveDirectoryIterator( $wp_content_dir, RecursiveDirectoryIterator::SKIP_DOTS );
+		$iterator = new RecursiveIteratorIterator( $directory, RecursiveIteratorIterator::CHILD_FIRST );
+
+		// Regex to match.
+		$this->regex = 'wp_cache_flush()';
+
+		foreach ( $iterator as $file ) {
+			$this->check_file( $file );
```
Comment (Sidsector9) [inline inc/checks/class-cache-flush.php]: @danielbachhuber , one concern is that this also works for string like "wp_cache_flush", which might be part of a comment.

Comment (danielbachhuber) [reply inc/checks/class-cache-flush.php]: Can you escape your regex so only `wp_cache_flush()` is found?

```
wp_cache_flush\(\)
```

File: inc/checks/class-cache-flush.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		}
+
+		if ( ! empty( $this->_matches ) ) {
+			$this->set_status( 'success' );
```
Comment (danielbachhuber) [inline inc/checks/class-cache-flush.php]: This status should be 'warning'

File: inc/checks/class-cache-flush.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+		// Path to wp-content directory.
+		$wp_content_dir = defined( 'WP_CONTENT_DIR' ) ? WP_CONTENT_DIR : ABSPATH . 'wp-content';
+		$directory = new RecursiveDirectoryIterator( $wp_content_dir, RecursiveDirectoryIterator::SKIP_DOTS );
+		$iterator = new RecursiveIteratorIterator( $directory, RecursiveIteratorIterator::CHILD_FIRST );
+
+		// Regex to match.
+		$this->regex = 'wp_cache_flush\(\)';
+
+		foreach ( $iterator as $file ) {
```
Comment (schlessera) [inline inc/checks/class-cache-flush.php]: Cleaner way:

```php
array_map( $iterator, array( $this, 'check_file' ) );
```

File: inc/checks/class-cache-flush.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+			$this->check_file( $file );
+		}
+
+		if ( ! empty( $this->_matches ) ) {
```
Comment (schlessera) [inline inc/checks/class-cache-flush.php]: Turn the conditions around for an early return, to get rid of the `else`.

File: inc/checks/class-cache-flush.php (hunk 4 above)
Comment (Sidsector9) [reply inc/checks/class-cache-flush.php]: `$iterator` is an object

**Comment by Sidsector9** [comment]
@schlessera I've made the changes

**Pull request merged**